pkg/staticpod/staticpod.go

```diff
@@ -280,22 +286,33 @@ func pod(args Args) (*v1.Pod, error) {
 		}
 	}
 
-	addVolumes(p, args.Dirs, true)
-	addVolumes(p, args.Files, false)
+	addVolumes(p, args.Sockets, socket)
+	addVolumes(p, args.Dirs, dir)
+	addVolumes(p, args.Files, file)
 
 	addExtraMounts(p, args.ExtraMounts)
 	addExtraEnv(p, args.ExtraEnv)
 
 	return p, nil
 }
 
-func addVolumes(p *v1.Pod, src []string, dir bool) {
+func addVolumes(p *v1.Pod, src []string, volume typeVolume) {
 	var (
-		prefix     = "dir"
-		sourceType = v1.HostPathDirectoryOrCreate
-		readOnly   = false
+		prefix     string
+		sourceType v1.HostPathType
+		readOnly   bool
 	)
-	if !dir {
+
+	switch volume {
+	case dir:
+		prefix = "dir"
+		sourceType = v1.HostPathDirectoryOrCreate
+		readOnly = false
+	case socket:
+		prefix = "sock"
+		sourceType = v1.HostPathSocket
+		readOnly = false
+	default:
 		prefix = "file"
 		sourceType = v1.HostPathFile
 		readOnly = true
```

Comment:
Just use the volumeType you passed in to define the prefix. No need to check if it matches the constants, then assign the same value as said constant.

```suggestion
	prefix = volume
	switch prefix {
	case dir:
		sourceType = v1.HostPathDirectoryOrCreate
		readOnly = false
	case socket:
		sourceType = v1.HostPathSocket
		readOnly = false
	case file:
		sourceType = v1.HostPathFile
		readOnly = true
```